### PR TITLE
chore: Add access_token support for the GCS remote state backend config

### LINKED ISSUE #2287: Add access_token support for the GCS remote state backend config
Hi!

Currently it does not seem to be possible to forward an access token to the GCS remote state backend:

```
remote_state {
  backend = "gcs"
  config = {
    # access_token = local.google_access_token # this doesn't work
    credentials = local.google_credentials # this works
    ...
  }
}
```

I see that credentials were added as a response to https://github.com/gruntwork-io/terragrunt/issues/780 but I think the support for access_token is still missing (it is supported by the Google provider otherwise, see https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#access_token).

Adding support for the access token would be awesome because then anyone would be able to provision with the gcloud credentials with this trick:

```
locals {
  google_access_token = run_cmd(
    "--terragrunt-quiet", "gcloud", "auth", "print-access-token",
    "--configuration", local.organization_id,
  )
}
```

Note that `credentials` isn't documented at the moment (see https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#remote_state, `gcs` backend section), it would be potentially great to document these too.

### PR DESCRIPTION
**Pull request by marcportabellaclotet-mt**
## Description

Fixes #2287.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

File: remote/remote_state_gcs.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ var terragruntGCSOnlyConfigs = []string{
 type RemoteStateConfigGCS struct {
 	Bucket        string `mapstructure:"bucket"`
 	Credentials   string `mapstructure:"credentials"`
+	AccessToken   string `mapstructure:"access_token"`
```
Comment (denis256) [inline remote/remote_state_gcs.go:55]: I think added field should be documented in GCS bucket reference `docs/_docs/04_reference/config-blocks-and-attributes.md`

File: remote/remote_state_gcs.go [hunk 2]
```diff
@@ -426,6 +427,11 @@ func CreateGCSClient(gcsConfigRemote RemoteStateConfigGCS) (*storage.Client, err
 
 	if gcsConfigRemote.Credentials != "" {
 		opts = append(opts, option.WithCredentialsFile(gcsConfigRemote.Credentials))
+	} else if gcsConfigRemote.AccessToken != "" {
```
Comment (denis256) [inline remote/remote_state_gcs.go:430]: I think will be helpful to have a test that will track if handling of `AccessToken` will continue to work in future

File: remote/remote_state_gcs.go (hunk 1 above)
Comment (marcportabellaclotet-mt) [reply remote/remote_state_gcs.go:55]: done

File: remote/remote_state_gcs.go (hunk 2 above)
Comment (marcportabellaclotet-mt) [reply remote/remote_state_gcs.go:430]: done

**Review by denis256 — APPROVED** [review]

**Pull request merged**